From 0a05fb7bcfa80dfed5dfcf11e7b4f3e624913c21 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Thu, 6 Aug 2020 10:55:52 +0800
Subject: [PATCH] drm/rockchip: cdn-dp use rockchip_drm_add_modes_noedid to get
 modes

use rockchip_drm_add_modes_noedid to get recommend modes when get
edid failed.

Change-Id: Iee9f0dc82c284654c363eaa2bb155654e51e0ba7
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/cdn-dp-core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/cdn-dp-core.c b/drivers/gpu/drm/rockchip/cdn-dp-core.c
index 30104912e249..10c07fc08522 100644
--- a/drivers/gpu/drm/rockchip/cdn-dp-core.c
+++ b/drivers/gpu/drm/rockchip/cdn-dp-core.c
@@ -278,6 +278,10 @@ static int cdn_dp_connector_get_modes(struct drm_connector *connector)
 		if (ret)
 			drm_connector_update_edid_property(connector,
 								edid);
+	} else {
+		ret = rockchip_drm_add_modes_noedid(connector);
+
+		dev_info(dp->dev, "failed to get edid\n");
 	}
 	mutex_unlock(&dp->lock);
 
-- 
2.35.3

